Packs/ApiModules/Scripts/GSuiteApiModule/GSuiteApiModule.py

```diff
@@ -0,0 +1,306 @@
+            raise DemistoException(error)
+
+    @staticmethod
+    def get_http_client(proxy: bool, verify: bool, timeout: int = 60) -> httplib2.Http:
+        """
+        Validate proxy and prepares Http object.
+
+        :param proxy: Boolean indicates whether to use proxy or not.
+        :param verify: Boolean indicates whether to use ssl certification.
+        :param timeout: Timeout value for request.
+
+        :return: ProxyInfo object.
+        :raises DemistoException: If there is any other issues while preparing proxy.
+        """
+        proxy_info = {}
+        proxies = handle_proxy()
+        if proxy:
+            https_proxy = proxies['https']
+            if not https_proxy.startswith('https') and not https_proxy.startswith('http'):
+                https_proxy = 'https://' + https_proxy
+            parsed_proxy = urllib.parse.urlparse(https_proxy)
+            proxy_info = httplib2.ProxyInfo(
+                proxy_type=httplib2.socks.PROXY_TYPE_HTTP,
+                proxy_host=parsed_proxy.hostname,
+                proxy_port=parsed_proxy.port,
+                proxy_user=parsed_proxy.username,
+                proxy_pass=parsed_proxy.password)
+
+        return httplib2.Http(proxy_info=proxy_info, disable_ssl_certificate_validation=verify, timeout=timeout)
+
+    @staticmethod
+    def validate_and_extract_response(response: Tuple[httplib2.Response, Any]) -> Dict[str, Any]:
+        """
+        Prepares an error message based on status code and extract a response.
+
+        :param response: Tuple containing httplib2.Response and content.
+
+        :return: response extracted json.
+        :raises DemistoException: If there is any other issues parsing response.
+        """
+        if response[0].status == 200 or response[0].status == 204:
+            return GSuiteClient.safe_load_non_strict_json(response[1])
+
+        status_code_message_map = {
+            400: COMMON_MESSAGES['BAD_REQUEST_ERROR'],
+            401: COMMON_MESSAGES['AUTHENTICATION_ERROR'],
+            403: COMMON_MESSAGES['AUTHORIZATION_ERROR'],
+            404: COMMON_MESSAGES['NOT_FOUND_ERROR'],
+            429: COMMON_MESSAGES['TOO_MANY_REQUESTS_ERROR'],
+            500: COMMON_MESSAGES['INTERNAL_SERVER_ERROR']
+        }
+
+        try:
+            # Depth details of error.
+            demisto.debug(response[1].decode() if type(response[1]) is bytes else response[1])
+            message = GSuiteClient.safe_load_non_strict_json(response[1]).get('error', {}).get('message', '')
+        except ValueError:
+            message = COMMON_MESSAGES['UNEXPECTED_ERROR']
+
+        if response[0].status in status_code_message_map:
+            raise DemistoException(status_code_message_map[response[0].status].format(message))
+        else:
+            raise DemistoException(COMMON_MESSAGES['UNKNOWN_ERROR'].format(response[0].status, message))
+
+    @staticmethod
+    def safe_load_non_strict_json(json_string: str) -> Dict[str, Any]:
```

Comment:
Similar functionality exists in `commonServerPython` in a function `safe_load_json`. 
Please us it if possible.

Reply:
@orhovy this method is basically used when someone copy and paste JSON secret key which contains /n char. Due to this, safe_load_json throws error so we used not only for test_module function but for other JSON safe loading.